shuffler/logic_to_aux.py

```diff
@@ -0,0 +1,85 @@
+import json
+from os import path
+from pathlib import Path
+import sys
+from typing import Any
+
+from _parser import clear_nodes, parse_area
+import click
+
+
+def logic_to_aux(logic_directory: str, output: str | None):
+    logic_files = list(Path(logic_directory).rglob("*.logic"))
+
+    for file in logic_files:
+        file_directory = file.relative_to(logic_directory)
+
+        rooms: dict[str, dict[str, Any]] = {}
+        with open(file, "r") as fd:
+            lines = [line.strip() for line in fd.readlines() if line and line.strip()]
+        clear_nodes()
+        nodes = parse_area(lines)
+
+        path_to_schema = "../"
+        directories_in_between = max(
+            str(file_directory).count("/"), str(file_directory).count("\\")
+        )
+
+        for _ in range(directories_in_between):
+            path_to_schema += "../"
+        path_to_schema += "aux-schema.json"
+
+        logic: dict[str, Any] = {
+            "$schema": str(path_to_schema),
+            "name": nodes[0].area,
+        }
+
+        for node in nodes:
+            room = rooms.get(node.room) or {}
+
+            chests = room.get("chests") or []
+            doors = room.get("doors") or []
```

Comment:
```suggestion
            room = rooms.get(node.room, {})

            chests = room.get("chests", [])
            doors = room.get("doors", [])
```
I believe this is slightly more idiomatic python. The second argument of the `get()` method for dictionaries is the value to default to if the key isn't in the dictionary